src/steps/check-google-analytics-page-view.ts

```diff
@@ -44,29 +44,32 @@ export class CheckGoogleAnalyticsPageView extends BaseStep implements StepInterf
                                         && r.url.includes('https://www.google-analytics.com')
                                         && r.url.includes('/collect')
                                         && r.url.includes('t=pageview')).map(r => r.url);
-      let actual = urls.filter(url => url.includes(`tid=${id}`));
+      const filteredRequest = urls.filter(url => url.includes(`tid=${id}`));
 
+      let records = [];
+      if (filteredRequest.length !== 1) {
+        let table;
+        if (filteredRequest.length > 1) {
+          table = this.createTable(filteredRequest);
+          records.push(table);
+        }
+        return this.fail('Expected 1 matching GA pageview, but %d matched.', [filteredRequest.length], records);
+      }
+
+      let filteredRequestsWithParams;
       if (Object.keys(expectedParams).length > 0) {
-        actual = actual.filter(u => this.includesParameters(u, expectedParams));
+        filteredRequestsWithParams = filteredRequest.filter(u => this.includesParameters(u, expectedParams));
       }
 
-      if (!isNullOrUndefined(actual[0])) {
-        params = this.getUrlParams(actual[0]);
+      if (!isNullOrUndefined(filteredRequestsWithParams[0])) {
```

Comment:
- [ ] This seems to cause a `TypeError: Cannot read property '0' of undefined`, but I think there are broader logical issues to resolve first.